libs-scala/resources/src/main/scala/com/digitalasset/resources/ProgramResource.scala

```diff
@@ -10,58 +10,55 @@ import com.daml.logging.LoggingContext.newLoggingContext
 import com.daml.resources.ProgramResource._
 
 import scala.concurrent.duration.{DurationInt, FiniteDuration}
-import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.{Await, ExecutionContext}
+import scala.util.Try
 import scala.util.control.{NoStackTrace, NonFatal}
-import scala.util.{Failure, Success, Try}
 
 class ProgramResource[T](
     owner: => ResourceOwner[T],
-    startupTimeout: FiniteDuration = 1.minute,
-    tearDownDuration: FiniteDuration = 10.seconds,
+    tearDownTimeout: FiniteDuration = 10.seconds,
 ) {
   private val logger = ContextualizedLogger.get(getClass)
 
   private val executorService = Executors.newCachedThreadPool()
-  private implicit val executionContext: ExecutionContext =
-    ExecutionContext.fromExecutor(executorService)
 
   def run(): Unit = {
     newLoggingContext { implicit logCtx =>
-      val resource = Try(owner.acquire()).fold(Resource.failed, identity)
+      val resource = {
+        implicit val executionContext: ExecutionContext =
+          ExecutionContext.fromExecutor(executorService)
```

Comment:
Probably simply passing the `executionContext` explicitly is simpler.

Reply:
Tried that but you end up with:

```scala
         Try(owner.acquire()(executionContext)).fold(Resource.failed(_)(executionContext), identity) 
```

That's a bit too noisy for me.